sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONRelation.scala

```diff
@@ -83,8 +80,10 @@ private[sql] class JSONRelation(
     val job = new Job(sqlContext.sparkContext.hadoopConfiguration)
     val conf = SparkHadoopUtil.get.getConfigurationFromJobContext(job)
 
-    val paths = inputPaths.map(_.getPath)
-
+    var paths = inputPaths.map(_.getPath)
+    if (paths.isEmpty) {
+      paths = cachedSourcePaths()   // try with original sources
```

Comment:
I am not so sure if the logic is correct, we will try to load the cached source path if the specified path is empty. Or do I missed something?